src/forward.rs

```diff
@@ -231,30 +247,23 @@ impl Forward {
         });
 
         loop {
-            let unix_stream = UnixStream::connect(self.socket.as_ref().unwrap()).await?;
-            let remote_stream = TcpStream::connect(&self.remote_addrs[0]).await?;
-
-            let peer_addr = remote_stream.peer_addr()?;
-            let unix_addr = unix_stream.peer_addr()?;
+            let unix_addr = self.socket.clone().unwrap();
+            let remote_addr = self.remote_addrs[0].clone();
 
-            let socket_path = unix_addr
-                .as_pathname()
-                .unwrap()
-                .to_str()
-                .unwrap()
-                .to_string();
+            let unix_stream = UnixStream::connect(&unix_addr).await?;
+            let remote_stream = TcpStream::connect(&remote_addr).await?;
 
-            info!("Connect to {} success", socket_path);
-            info!("Connect to {} success", peer_addr);
+            info!("Connect to {} success", unix_addr);
+            info!("Connect to {} success", remote_addr);
 
             let connector = connector.clone();
 
             let unix_stream = tcp::NetStream::Unix(unix_stream);
             let remote_stream = tcp::NetStream::from_connector(remote_stream, connector).await;
 
-            info!("Open pipe: {} <=> {}", socket_path, peer_addr);
+            info!("Open pipe: {} <=> {}", unix_addr, remote_addr);
             tcp::handle_forward(unix_stream, remote_stream).await?;
-            info!("Close pipe: {} <=> {}", socket_path, peer_addr);
+            info!("Close pipe: {} <=> {}", unix_addr, remote_addr);
```

Comment:
the connection will still be blocked when multi connections are coming in `socket_to_remote` function